pkg/httpc/client.go

```diff
@@ -113,22 +109,27 @@ func (c *Client) Send(
 		return nil, e
 	}
 
-	if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
+	var data []byte
+	if !c.isRawBody {
+		data, err = ioutil.ReadAll(resp.Body)
 		defer resp.Body.Close()
-		data, _ := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			e := errType.Wrap(err, "Failed to read %s API response", errOriginComponent)
+			log.Warn("SendRequest failed", zap.String("uri", uri), zap.Error(err))
+			return nil, e
+		}
+	}
+
+	if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
 		e := errType.New("Request failed with status code %d from %s API: %s", resp.StatusCode, errOriginComponent, string(data))
 		log.Warn("SendRequest failed", zap.String("uri", uri), zap.Error(err))
 		return nil, e
 	}
 
-	return &Response{resp}, nil
+	return &Response{RawResponse: resp, Body: data}, nil
 }
 
 type Response struct {
-	*http.Response
-}
-
-func (r *Response) Body() ([]byte, error) {
-	defer r.Response.Body.Close()
-	return ioutil.ReadAll(r.Response.Body)
+	RawResponse *http.Response
+	Body        []byte
```

Comment:
This API design of worse compared to the previous one. As an example,

suppose you see a piece of code (actually everywhere in the code):

```
...(resp.Body)
```

Can you tell out what kind of `resp.Body` it is? And how can we identify its correctness? You simply can't. Since there is ambiguity between `http.Response.Body` and `httpc.Response.Body`. Not to mention that `http.Response` and `httpc.Response` is passed everywhere (e.g. `resp, err := xxxx()`), which causes a new ambiguity.

Reply:
Yes! I should unify the response type.